internal/juju/applications.go

```diff
@@ -187,22 +190,11 @@ func (c applicationsClient) CreateApplication(input *CreateApplicationInput) (*C
 	if err != nil {
 		return nil, err
 	}
+	defer func() { _ = conn.Close() }()
 
 	charmsAPIClient := apicharms.NewClient(conn)
-	defer charmsAPIClient.Close()
```

Comment:
I wrote it up in the commit message.  All of the api clients share a connection, if you close one client, the connection closes as well. The code only worked because all of the clients were created at once, and the defer Close() called, even if the client wasn't used. When I moved to create and close clients when they were used, the code stopped working. Double checked the juju api code for same.